v3/generator/client/client.tmpl

```diff
@@ -134,6 +149,21 @@ func NewClient(credentials *credentials.Credentials, opts ...ClientOpt) (*Client
 	return client, nil
 }
 
+// getUserAgent only for compatibility with UserAgent.
+func (c *Client) getUserAgent() string {
+	defaultUA := getDefaultUserAgent()
+
+	if c.userAgent != defaultUA {
+		return c.userAgent
+	}
+
+	if UserAgent != defaultUA {
+		return UserAgent
+	}
+
+	return c.userAgent
+}
+
 // WithEndpoint returns a copy of Client with new zone Endpoint.
 func (c *Client) WithEndpoint(endpoint Endpoint) *Client {
```

Comment:
Thanks! Can you also update all the `WithX` to make sure user agent is copied?
Feel free to add a:
```Golang
func (c *Client) WithUserAgent(userAgent string) *Client {
```

`WithX` method are useful when we want to return a copy of the client with a new option
```Golang
client2 := client.WithUserAgent("test user-agent").WithEnpoint(x).....
```